### PR TITLE
Improve CSS

### PR DESCRIPTION
**Pull request by cindyli**
This pull request does:

1. Add the mobile styling for the hospital description panel;
2. Coordinates with https://github.com/inclusive-design/wecount.inclusivedesign.ca/pull/415 to have map responsive design work properly with the WeCount website;
3. Add functional code for expand/collapse buttons.

### CODE REVIEW

File: src/js/expandCollapseButton.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+fluid.bindExpandCollapseButtonEvents = function (that) {
+    that.locate("expandCollapseButton").click(function () {
+        that.applier.change("expanded", !that.model.expanded);
+    });
+};
+
+fluid.toggleExpandedState = function (expandCollapseButton, expanded, elementsToToggle) {
+    console.log(expandCollapseButton);
+    expandCollapseButton[0].setAttribute("aria-expanded", expanded);
+    expandCollapseButton[0].setAttribute("aria-label", expanded ? "expand" : "collapse");
```
Comment (amb26) [inline src/js/expandCollapseButton.js]: I think these labels are applied the wrong way round. Given they are phrased as active verbs, "expand", I think they should reflect the action which will be performed when operating the button rather than labelling its state. I've reversed them in my pull

File: src/js/expandCollapseButton.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+fluid.defaults("fluid.expandCollapseButton", {
+    gradeNames: "fluid.viewComponent",
+    selectors: {
+        expandCollapseButton: ".fl-mapviz-expand-collapse-button"
+    },
+    model: {
+        expanded: true
+    },
+    invokers: {
+        toggleExpandedState: {
```
Comment (amb26) [inline src/js/expandCollapseButton.js]: I've reworked this component to make it fully integral and removing the need for invokers and overrides like this in my branch https://github.com/cindyli/covid-data-monitor/pull/1

**Comment by amb26** [comment]
I've issued a pull at https://github.com/cindyli/covid-data-monitor/pull/1 which reworks the expand buttons to be integral, as well as removing all the "fluid.activatable" integration which is no longer required since you've rewritten all the markup so that it is native activatable.
I believe that my branch behaves identically to yours, but I see that there are several interaction problems created by the new buttons. Primarily - 
i) It is now possible to hide the filter panel so that it can never be restored - I think this is caused by the markup being nested the wrong way, but I could be wrong
ii) It is possible to hide controls in general so they can never be restored by hiding them in the mobile view and then expanding the browser window. I imagine this is the issue you wanted to talk about with respect to adding a CSS-to-JS context detection. Presumably we will fix this by automatically expanding all collapsed controls when we detect a transition to the desktop view.

**Comment by cindyli** [comment]
@amb26 the latest commit uses the context awareness to distinguish the desktop or the mobile view and apply proper component accordingly. 

Now I hit the issue of how to handle the transition between the desktop and the mobile views. Using an example when the transition moves from the desktop view to the mobile view, I think an ideal solution is to automatically replace the `desktopViewHandler` with the `mobileViewHandler` then get the new handler in action. However, I'm not sure if this is possible. Even if it's possible, I'm not sure if the new handler component can take effect without a re-instantiation of the component tree. @amb26, you talked about using `resolveRoot` the other day. Can you elaborate on it if that's a possible solution? If I didn't rmb correctly, do you have thoughts on this issue? Thanks.

File: src/js/covidMap.js [hunk 3]
```diff
@@ -338,21 +414,16 @@ fluid.defaults("fluid.covidMap.map", {
...
-                        changePath: "{map}.model.query",
-                        value: ""
-                    }
-                }
-            }
+        searchResultsBackButton: {
+            type: "fluid.backButton",
+            container: "{that}.dom.searchResultsBackButton"
+        },
+        hosptialBackButton: {
```
Comment (amb26) [inline src/js/covidMap.js]: Typo here hosptial

File: src/js/covidMap.js [hunk 4]
```diff
@@ -441,6 +508,182 @@ fluid.defaults("fluid.covidMap.map", {
...
+        locationButtonOnMobile: {
+            type: "fluid.button",
+            container: "{that}.dom.locationButtonOnMobile",
+            options: {
+                modelListeners: {
+                    "showCitiesList": {
+                        path: "activate",
+                        changePath: "{map}.model.citiesListVisible",
+                        value: true,
+                        excludeSource: ["init"]
```
Comment (amb26) [inline src/js/covidMap.js]: Lots of excludeSource: ["init"] annotations in new-style code are a code smell that the interaction hasn't been encoded in a properly integral way and should aim to be removed

File: src/js/covidMap.js [hunk 5]
```diff
@@ -441,6 +508,182 @@ fluid.defaults("fluid.covidMap.map", {
...
+    modelRelay: {
+        filterPanelVisibilityEffect: {
+            target: "{that}.model",
+            singleTransform: {
+                type: "fluid.transforms.valueMapper",
+                defaultInput: "{that}.model.filterPanelVisible",
+                match: {
+                    true: {
+                        outputValue: {
+                            filterPanelExpanded: true,
```
Comment (amb26) [inline src/js/covidMap.js]: I'm struggling to get hold of the model behind the pattern of flags - it seems like the rule is something like "make the just selected panel visible whilst hiding the others" and I think this could better be expressed by transmitting the relations through a String encoding "visiblePanel": "filter"/"results"/"cities" etc and then separately running a tiny literal modelRelay rule outputting a boolean like so
````
multiplexFilters: {
    source: "visiblePanel",
    func: panel => panel === "filters",
    target: "{map}.model.dom.filterPanel.visible"
}
````
Note that now the visibility/expandedness of the panel is already directly materialised in the model you don't need to allocate separate flags such as "filterPanelVisible" etc. to name them. Also note that we are moving towards the model of "tiny literal functions" rather than the old-fashioned bulky JSON blocks - although the valueMapper is one of the transforms that will probably end up surviving in the the new world, but for different/more complex use cases than this one

File: src/js/covidMap.js [hunk 6]
```diff
@@ -64,8 +87,29 @@ fluid.covidMap.extractPostcodes = function (rows, field) {
...
+};
+
+fluid.contextAware.makeChecks({
+    "fluid.isMobileView": "fluid.covidMap.isMobileView"
+});
+
 fluid.defaults("fluid.covidMap.map", {
-    gradeNames: ["hortis.leafletMap", "hortis.streetmapTiles", "hortis.CSVLeafletMap", "hortis.conditionalTemplateRenderer"],
+    gradeNames: ["hortis.leafletMap", "hortis.streetmapTiles", "hortis.CSVLeafletMap", "hortis.conditionalTemplateRenderer", "fluid.contextAware"],
+    contextAwareness: {
```
Comment (amb26) [inline src/js/covidMap.js]: contextAwareness isn't a good choice in this case since as you note in the comment it can only apply grades in a static way. There may be an equivalent feature in the upcoming framework that is more dynamic, but for this use case I recommend the following pattern:
i) A top-level listener attached to your MediaQuery via https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener which sets a boolean flag in a model at the root
ii) Relay the flag and the negation of the flag into two booleans in the map component, e.g. useMobileView and useDesktopView
iii) Use two new-style boolean lensed components which attach these flags to subcomponents, e.g.
````
dynamicComponents: {
    mobileView: {
        source: "{map}.model.useMobileView",
        type: "fluid.covidMap.mobileViewHandler"
    }
````
etc.
The definitions will then need to be rewritten so that they function as subcomponents via relay definitions to the parent model rather than as mixin grades. We will probably then run into some model state coordination bugs that we will either fix or paper over somehow : P

Comment (amb26) [reply src/js/covidMap.js]: You can see the test case showing how this facility works at https://github.com/fluid-project/infusion/pull/1038/files#diff-6972ab970b7ccb5a497ebc0fc51a19df59946de2ba24f58b1577c4ae40546f8cR1005-R1036

Comment (amb26) [reply src/js/covidMap.js]: Updated comment above to read "dynamicComponents" rather than "components" - my mind is already in the future when there will only be "regular effing components" : P

**Comment by cindyli** [comment]
@amb26, thanks for the suggestion and the fantastic new data binding feature for constructing dynamic components. It does simplify the handling of the transition between the desktop and the mobile views.

This pull request is ready for another review. Thanks.

File: src/js/covidMap.js [hunk 7]
```diff
@@ -437,6 +537,97 @@ fluid.defaults("fluid.covidMap.map", {
...
+                citiesList: true,
+                resultsPage: true,
+                filterPanel: true,
+                hospitalPanel: true
+            }]
+        }
+    },
+    modelRelay: {
+        citiesShowing: {
+            source: "resultsShowing",
```
Comment (amb26) [inline src/js/covidMap.js]: I think this rule will be ineffective since neither the source nor target are on this component

File: src/js/covidMap.js [hunk 8]
```diff
@@ -437,6 +537,97 @@ fluid.defaults("fluid.covidMap.map", {
...
+    mediaQuery.addListener(function (evt) {
+        that.applier.change("isDesktopView", evt.matches ? true : false);
+    });
+};
+
+// The desktop view handler
+fluid.defaults("fluid.covidMap.desktopViewHandler", {
+    gradeNames: "fluid.modelComponent",
+    listeners: {
+        "onCreate.setVisiblePanels": {
```
Comment (amb26) [inline src/js/covidMap.js]: For both of these viewHandlers it will be clearer to relay both visiblePanels and expandedPanels as a whole into its model, and then provide these "onCreate" arguments as initial model values. Hopefully it will work!

File: src/js/covidMap.js [hunk 9]
```diff
@@ -422,6 +482,46 @@ fluid.defaults("fluid.covidMap.map", {
         filterCountOnMobile: {
             type: "fluid.covidMap.filterCount",
             container: "{that}.dom.filterCountOnMobile"
+        },
+        locationButtonOnMobile: {
```
Comment (amb26) [inline src/js/covidMap.js:520]: Perhaps the definition of these buttons could be moved into the mobile view handler?
The fact that "excludeSource" is required is I think a bug in the "fluid.button" implementation which we haven't run into so far since the buttons that we have so far were idempotent.
The 2nd definition looks incorrect since as I understand it the "expandedPanels" is a hash to boolean rather than an array.
I think we're getting to the point at which we might require some at least basic test cases - or at the least, a manual testing script instructions so that someone can quickly run through the app and verify that it is behaving as intended.

File: src/js/covidMap.js [hunk 10]
```diff
@@ -267,6 +302,32 @@ fluid.defaults("fluid.covidMap.map", {
...
             },
             args: ["{that}.model.zoom", "{that}.options.smallMarkersBelowZoom"]
+        },
+        isMobileView: {
+            source: "isDesktopView",
+            target: "isMobileView",
+            func: value => !value
+        },
+        // Panel visibilities
+        citiesListVisibility: {
```
Comment (amb26) [inline src/js/covidMap.js]: This block would probably work better as a fluid.transforms.arrayToSetMembership transform, acting on the panels visibility held in both forms - e.g. visiblePanelArray and visiblePanelFlags. You can then write explicit definitions coupling all the flags - e.g.
visiblePanelFlags: {
    citiesList: "{that}.model.dom.citiesList.visible"
}
etc.

File: src/js/covidMap.js [hunk 11]
```diff
@@ -437,6 +537,97 @@ fluid.defaults("fluid.covidMap.map", {
...
+        mobileViewHandler: {
+            source: "{map}.model.isMobileView",
+            type: "fluid.covidMap.mobileViewHandler"
+        }
+    }
+});
+
+fluid.covidMap.detectWindowSize = function (that) {
+    const mediaQuery = window.matchMedia(that.options.mediaQueryBreakpoint);
+    that.applier.change("isDesktopView", mediaQuery.matches ? true : false);
```
Comment (amb26) [inline src/js/covidMap.js]: matches is already a boolean so the ternary is not necessary on these two lines

File: src/js/covidMap.js [hunk 12]
```diff
@@ -583,27 +788,67 @@ fluid.defaults("fluid.styledButton", {
...
+    components: {
+        expandButton: {
+            type: "fluid.expandButton",
+            container: "{citiesList}.dom.expandButton",
+            options: {
+                model: {
+                    expanded: "{map}.model.expandedPanels.citiesList"
+                },
+                modelRelay: {
+                    collapseTitle: {
```
Comment (amb26) [inline src/js/covidMap.js]: In the other expandButton these rules were prefixed "visible" rather than "collapse"

File: src/js/covidMap.js [hunk 13]
```diff
@@ -669,6 +915,23 @@ fluid.defaults("fluid.covidMap.resultsPage", {
...
+    components: {
+        expandButton: {
+            type: "fluid.expandButton",
+            container: "{resultsPage}.dom.expandButton",
+            options: {
+                model: {
+                    expanded: "{map}.model.expandedPanels.resultsPage"
+                },
+                modelRelay: {
+                    collapseResultsList: {
```
Comment (amb26) [inline src/js/covidMap.js]: See above

File: src/js/covidMap.js [hunk 14]
```diff
@@ -437,6 +537,97 @@ fluid.defaults("fluid.covidMap.map", {
...
+            target: "{map}.model.visiblePanels",
+            func: x => {if (x) {return ["resultsPage"];}}
+        },
+        expandResultsPage: {
+            source: "{map}.model.resultsShowing",
+            target: "{map}.model.expandedPanels.resultsPage",
+            func: x => {if (x) {return true;}}
+        },
+        onlyShowHospitalPanel: {
+            source: "{map}.model.selectedIndex",
```
Comment (amb26) [inline src/js/covidMap.js]: It's somewhat hard to follow what is going on here... especially since there is a different style of representation for the expansion and visibility of the panels. The comment at the head of this block doesn't seem fully accurate. 
It looks something like "When there are results, expand and show the resultsPage only"?

File: src/js/covidMap.js (hunk 8 above)
Comment (cindyli) [reply src/js/covidMap.js]: @amb26, your suggestion on [FLUID-6603](https://issues.fluidproject.org/browse/FLUID-6603) makes it possible to relay the initial model values of `visiblePanelArray` and `expandedPanels` to the parent component. But I ran into another issue with these values being set incorrectly after the construction of the conditional sub-component. I've been trying to isolate the problem in order to create a test case but had no success. So I pushed up the code change in this branch and hope you can reproduce the problem.

Using the mobile sized window is easier to reproduce the problem.

**Expected initial mobile view**: The cities list is visible and collapsed. All other panels are hidden.
expected

**Problem with relaying initial model values from the conditional sub-component**: all panels are hidden.
problem

**The finding from the debug***
1. [Tracking model changes](https://github.com/cindyli/covid-data-monitor/blob/improve-css/src/js/covidMap.js#L385-L399) on `{map}.model.visiblePanelArray` and `{map}.model.expandedPanels` shows, before the window size is detected and `mobileViewHandler` is constructed, they are set to some initial values by some other `modelRelay` actions;
2. When the window size is detected and `mobileViewHandler` is constructed, `{map}.model.visiblePanelArray` and `{map}.model.expandedPanels` are then relayed to the initial model values of `mobileViewHandler`;
3. However, when debugging `{map}.model.visiblePanelArray` and `{map}.model.expandedPanels` at the `mobileViewHandler` onCreate event, these values remain as what are set in step 1.
4. Nevertheless, as the problematic screenshot shows, none of these model values are applied so none of panels are visible.

If step 2 uses `mobileViewHandler` onCreate event to call `{map}.applier.change`, it will set `visiblePanelArray` and `expandedPanels` correctly. The debug at step 3 confirms it.

@amb26, please let me know if my steps are not clear. Thanks.

Comment (cindyli) [reply src/js/covidMap.js]: @amb26, as we talked at today's meeting, since using "onCreate" event to set initial model value is a working work-around, I reverted to this approach from relaying the initial model values that are set directly on the `model` block.

This branch is ready for another review and merge. 

My next step is to update [the wecount website PR](https://github.com/inclusive-design/wecount.inclusivedesign.ca/pull/415) to fetch the map from this branch.

File: src/js/expandButton.js [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+            singleTransform: "fluid.transforms.toggle"
+        },
+        ariaExpanded: {
+            source: "expanded",
+            target: "dom.container.attrs.aria-expanded"
+        },
+        ariaLabel: {
+            source: "expanded",
+            target: "dom.container.attrs.aria-label",
+            func: expanded => expanded ? "collapse" : "expand"
```
Comment (greatislander) [inline src/js/expandButton.js]: My understanding ([based on recent reading](https://kittygiraudel.com/2021/03/13/gorillas-nav-a-case-study/#shut-it-down)) is that because the `aria-expanded` attribute indicates state, this label should be static and should explain _what_ is being collapsed/expanded. Currently when interacting with this component, VoiceOver reads:

"expand, collapsed, button"

…and then:

"collapse, expanded, button"

Which doesn't tell me anything about what the button is actually controlling.

**Review by greatislander — COMMENTED** [review]
I noticed that this uses an event listener to detect width and apply responsive layout. I think there is probably a way to do this with CSS and media queries alone. Here is a demo:

https://codepen.io/greatislander/full/bGgNMXW

This is overly simplified but basically the trick is that the mobile visibility toggle is based on the presence of a class, which means that the visibility can be overridden in the CSS without the need to detect a resize event.

EDIT: One other note is that it's currently possible for state to be lost when resizing a browser window or rotating a tablet (I was able to reproduce this on my iPad by using the locations filter to find a particular clinic, then rotating from portrait to landscape or vice versa). This seems like undesireable behaviour.

**Comment by cindyli** [comment]
@greatislander, @amb26, just addressed @greatislander's comment to use CSS to control the transition between the desktop and mobile views. The solution is to use [a special css class](https://github.com/cindyli/covid-data-monitor/blob/improve-css/src/css/fluid-covid-map-viz.css#L767-L774) that's only available on the mobile view to control panel visibility on the mobile view. It doesn't affect the desktop at all. This approach introduces another benefit that panel states are maintained when the screen size moves from the desktop to mobile and vice versa.

This pull request is ready for another review.

File: src/js/covidMap.js [hunk 16]
```diff
@@ -420,34 +499,78 @@ fluid.defaults("fluid.covidMap.map", {
...
-            container: "{that}.dom.filterCountOnDesktop"
-        },
-        filterCountOnMobile: {
-            type: "fluid.covidMap.filterCount",
-            container: "{that}.dom.filterCountOnMobile"
+        filterButtonOnMobile: {
+            type: "fluid.button",
+            container: "{map}.dom.filterButtonOnMobile",
+            options: {
+                modelRelay: {
```
Comment (amb26) [inline src/js/covidMap.js]: This is more idiomatically a changePath listener - 
e.g. 
````
modelListeners: {
    "showFilterPanel: {
        source: "activate",
        changePath: "{map}.model.visiblePanelOnMobileFlags",
        value: {
            "citiesList": false,
            "resultsPage": false,
            "filterPanel": true,
            "hospitalPanel": false
      },
      excludeSource: "init"
````
Given the mutual exclusion between the flags it would be better to model this as a single selection field plus a relay to the flags but unfortunately we don't have such a transform currently implemented - fluid.transforms.arrayToSetMembership expects a multiple selection. I've written up this issue as https://issues.fluidproject.org/browse/FLUID-6612

File: src/js/covidMap.js [hunk 17]
```diff
@@ -420,34 +499,78 @@ fluid.defaults("fluid.covidMap.map", {
...
+                                    "citiesList": true,
+                                    "resultsPage": false,
+                                    "filterPanel": false,
+                                    "hospitalPanel": false
+                                };
+                            };
+                        }
+                    },
+                    expandCitiesList: {
+                        source: "activate",
```
Comment (amb26) [inline src/js/covidMap.js]: As below this is better as
````
    modelListeners: {
        expandCitiesList: {
            path: "activate",
            changePath: "{citiesList}.expandButton.model.expanded",
            value: true,
            excludeSource: "init"
        }
````

File: src/js/covidMap.js [hunk 18]
```diff
@@ -257,14 +297,33 @@ fluid.defaults("fluid.covidMap.map", {
...
             source: "resultsShowing",
             target: "dom.resultsPage.visible"
         },
-        citiesShowing: {
+        onlyShowResultsPanelOnMobile: {
             source: "resultsShowing",
-            target: "dom.citiesList.visible",
-            func: x => !x
+            target: "visiblePanelOnMobileFlags",
+            func: x => {
```
Comment (amb26) [inline src/js/covidMap.js:317]: See below for rationalising these as modelListeners

File: src/js/expandButton.js [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+
+"use strict";
+
+// The expand-collapse button component. This button is only available on the mobile view.
+// This component adds or removes a mobile-view-only visibility css class to a given list of DOM elements
+// based on the expanded state.
+fluid.defaults("fluid.expandButton", {
+    gradeNames: "fluid.button",
+    // Supplied by integrators. An array of DOM element to expand or collapse.
+    domElements: [],
```
Comment (amb26) [inline src/js/expandButton.js]: More descriptive name like "elementsToExpand"

File: tests/manual/desktopViewTestPlan.md [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+1. Select a number of checkboxes in this panel. While checking, the number at the top right corner of the panel
+increases to show the total number of selected filters;
+2. Deselect some checkboxes, the number at the top right corner of the panel decreases;
+3. Make sure some checkboxes are checked;
+4. Click "reset filters" button;
+5. All checkboxes are unchecked. the number at the top right corner of the panel restores to 0.
+
+### Apply filters
+
+This feature is not yet implemented. It will be implemented when the accessibility data for COVID assessment centres
```
Comment (amb26) [inline tests/manual/desktopViewTestPlan.md]: If this is merged up with main, the feature is implemented with respect to the placeholder data

File: tests/manual/desktopViewTestPlan.md [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+is ready.
+
+## Map zoom in & zoom out
+
+Users can zoom the map by clicking on the zoom in or the zoom out buttons.
```
Comment (amb26) [inline tests/manual/desktopViewTestPlan.md]: As well as the mouse wheel

File: src/js/covidMap.js [hunk 22]
```diff
@@ -955,3 +1155,42 @@ fluid.covidMap.addMarkers = function (that) {
...
+        toggleCssClass: {
+            path: "visible",
+            func: "fluid.covidMap.visiblePanel.toggleClass",
+            args: ["{that}"]
+        }
+    }
+});
+
+fluid.covidMap.visiblePanel.toggleClass = function (that) {
+    // Use "null" as the initial visibility value in order not to programmatically apply any visibility control
```
Comment (amb26) [inline src/js/covidMap.js:1184]: I'll investigate what is clearly a framework bug

**Comment by cindyli** [comment]
@amb26: all comments are addressed and ready for another look.

File: src/js/covidMap.js [hunk 23]
```diff
@@ -221,12 +248,25 @@ fluid.defaults("fluid.covidMap.map", {
...
         isHospitalSelected: {
             target: "dom.hospitalPanel.visible",
             source: "selectedIndex",
             func: "fluid.isValue"
         },
+        onlyShowHospitalPanelOnMobile: {
+            target: "visiblePanelOnMobileFlags",
+            source: "selectedIndex",
+            func: x => {
+                if (!!x) {
```
Comment (amb26) [inline src/js/covidMap.js]: Is this really right? Seems like it will treat an index of 0 as a missing value. You should use the same check as https://github.com/inclusive-design/covid-data-monitor/blob/8422fd6f800c4365fe222c2b9e2c29ffc58bb40c/src/js/covidMap.js#L251 - or perhaps factor these both into a dedicated isHospitalShowing model flag that both definitions can use

File: src/js/covidMap.js (hunk 18 above)
Comment (amb26) [reply src/js/covidMap.js:317]: This comment is still unaddressed

File: src/js/covidMap.js [hunk 24]
```diff
@@ -609,27 +731,72 @@ fluid.defaults("fluid.styledButton", {
...
+            changePath: "{map}.model.query",
+            value: ""
+        },
+        "resetSelectedIndex": {
+            path: "activate",
+            changePath: "{map}.model.selectedIndex",
+            value: null
+        },
+        "showCitiesList": {
+            path: "activate",
```
Comment (amb26) [inline src/js/covidMap.js:758]: This should probably have an "excludeSource: init" on it

File: src/js/covidMap.js (hunk 18 above)
Comment (cindyli) [reply src/js/covidMap.js:317]: I think this model relay is different from other ones such as [`expandCitiesList`](https://github.com/inclusive-design/covid-data-monitor/blob/8422fd6f800c4365fe222c2b9e2c29ffc58bb40c/src/js/covidMap.js#L531-L536) for the location button that can be simplified using modelListeners with `excludeSource: "init"` as the location button doesn't need to handle a false scenario once the initial model change is skipped.

This model relay is controlled by `resultsShowing`, a boolean value that is flipped back and forth when a user interacts with UI. This function ensures the result panel is only shown when `resultsShowing === true` and nothing changes when `resultsShowing === false`. Even if rewriting it into a model listener, the same check and the same function remain without any simplification.

Please let me know if I misunderstood your point. Thanks.
